File: PMS.BO/TimeTracking/TimeTracking.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PMS.BO
{
	public class TimeTracking
	{
		#region Constractor
		public TimeTracking() { }
		#endregion
		public int TimeTrackingId { get; set; }
		public DateTime StartTime { get; set; }
		public DateTime EndTime { get; set; }
		public string Notes { get; set; } = string.Empty;

		// Calculated Property (Not mapped to DB)
		public TimeSpan Duration => EndTime - StartTime;

		// Foreign Keys
		public int UserId { get; set; }
		public int TaskId { get; set; }

		// Navigation Properties
		public User User { get; set; } = null!;
		public ProjectTask Task { get; set; } = null!;
	}
}

File: PMS/ClientApp/src/app/PMS_MODEL/TimeTracking/TimeTracking.ts
import { BasicBaseObject } from "../Basic/BasicBaseObject";

export class TimeTracking extends BasicBaseObject{
    constructor(){
        super();
      this.startTime = "";
      this.endTime = "";
      this.notes = "";
      this.userId = 0;
      this.taskId = 0;
    }
  startTime: string;
  endTime: string;
  notes: string;
  userId: number;
  taskId: number;
}


